v2.0/partitioning.md

```diff
@@ -0,0 +1,371 @@
+~~~
+
+### Define Table Partitions by Range
+
+Consider the table of students of the global online learning portal, RoachLearn. Suppose we want to store the data of current students on fast and expensive storage devices (example: SSD) and store the data of the graduated students on slower, cheaper storage devices(example: HDD). We can achieve this by partitioning the table by date and using the `PARTITION BY RANGE` syntax. 
+
+#### Step 1: Set the enterprise license
+
+To set the enterprise license, see [Set the Trial or Enterprise License Key](enterprise-licensing.html#set-the-trial-or-enterprise-license-key)
+
+#### Step 2: Start each node with the appropriate storage device specified in the `--store` flag:
+
+~~~ shell
+$ cockroach start --insecure --attrs=ssd --store=node1 --host=<node1 hostname> --port=26257 --http-port=8080 --join=<node1 hostname>:26257,<node2 hostname>:26258
+
+$ cockroach start --insecure --attrs=hdd --store=node2 --host=<node2 hostname> --port=26258 --http-port=8081 --join=<node1 hostname>:26257,<node2 hostname>:26258
+~~~
+
+#### Step 3: Create a table with the appropriate partitions:
+ 
+~~~ sql
+> CREATE TABLE students_by_range (
+   id SERIAL,
+   name STRING,
+   email STRING,                                                                                           
+   country STRING, 
+   expected_graduation_date DATE,                                                                                      
+   PRIMARY KEY (expected_graduation_date, id)) 
+   PARTITION BY RANGE (expected_graduation_date) 
+      (PARTITION graduated VALUES FROM (MINVALUE) TO ('2017-08-15'), 
+      PARTITION current VALUES FROM ('2017-08-15') TO (MAXVALUE));
+~~~
+
+#### Step 4: Create corresponding zone configurations:
+ 
+~~~ shell
+$ cat > current.zone.yml
+constraints: [+ssd]
+
+$ cat > graduated.zone.yml
+constraints: [+hdd]
+~~~
+
+#### Step 5: Apply zone configurations to corresponding partitions:
+ 
+~~~ shell
+$ cockroach zone set roachlearn.students_by_range.current --insecure  -f current.zone.yml
+$ cockroach zone set roachlearn.students_by_range.graduated --insecure  -f graduated.zone.yml
+~~~
+
+### Define Subpartitions on a Table
+
+A list partition can itself be partitioned, forming a subpartition. There is no limit on the number of levels of subpartitioning; that is, list partitions can be infinitely nested.
+
+{{site.data.alerts.callout_info}}Range partitions cannot be subpartitioned.{{site.data.alerts.end}}
+
+Going back to RoachLearn's scenario, suppose we want to do all of the following:
+
+- Keep the students' data close to their location.
+- Store the current students' data on faster storage devices.
+- Store the graduated students' data on slower, cheaper storage devices (example: HDD).
+
+We can achieve this by partitioning the table first by location and then by date.
+
+#### Step 1: Start each node with the appropriate storage device specified in the `--store` flag:
```

Comment:
Ah, now I see you using store-level attributes. Perhaps this is what you wanted to do in the example above as well?